packages/features/ee/payments/components/Payment.tsx

```diff
@@ -173,4 +177,52 @@ export default function PaymentComponent(props: Props) {
       )}
     </form>
   );
+};
+
+const ELEMENT_STYLES: stripejs.Appearance = {
+  theme: "none",
+  rules: {
+    ".Label": {
+      fontSize: "16px",
+    },
+  },
+};
+
+const ELEMENT_STYLES_DARK: stripejs.Appearance = {
+  theme: "night",
+  variables: {
+    colorText: "#d6d6d6",
+    fontWeightNormal: "600",
+    borderRadius: "6px",
+    colorBackground: "#101010",
+    colorPrimary: "#d6d6d6",
+  },
+};
+
+export default function PaymentComponent(props: Props) {
+  const stripePromise = getStripe((props.payment.data as StripePaymentData).stripe_publishable_key);
+  const paymentOption = props.payment.paymentOption;
+  const [darkMode, setDarkMode] = useState<boolean>(false);
+  let clientSecret;
+
+  useEffect(() => {
+    setDarkMode(window.matchMedia("(prefers-color-scheme: dark)").matches);
+  }, []);
+
+  if (paymentOption === "HOLD" && "setupIntent" in props.payment.data) {
+    clientSecret = props.payment.data.setupIntent.client_secret;
+  } else if (!("setupIntent" in props.payment.data)) {
+    clientSecret = props.payment.data.client_secret;
+  }
+
+  return (
+    <Elements
+      stripe={stripePromise}
+      options={{
+        clientSecret: clientSecret!,
+        appearance: darkMode ? ELEMENT_STYLES_DARK : ELEMENT_STYLES,
+      }}>
+      <PaymentForm {...props} />
+    </Elements>
```

Comment:
Extract the Stripe `Elements` to its own component if in the future we add more payment apps.